@celiayz  When I run the below example query given in the github page, I am getting the following `INFO:  Bin count threshold was too large to find approximate bounds. Either run over a larger dataset or decrease success_probability and try again. Returning NULL.`


```
SELECT result.fruit, result.number_eaten
FROM (
  SELECT per_person.fruit,
    ANON_SUM(per_person.fruit_count, LN(3)/2) as number_eaten,
    ANON_COUNT(uid, LN(3)/2) as number_eaters
    FROM(
      SELECT * , ROW_NUMBER() OVER (
        PARTITION BY uid
        ORDER BY random()
      ) as row_num
      FROM (
        SELECT fruit, uid, COUNT(fruit) as fruit_count
        FROM FruitEaten
        GROUP BY fruit, uid
      ) as per_person_raw
    ) as per_person
  WHERE per_person.row_num <= 5
  GROUP BY per_person.fruit
) as result
WHERE result.number_eaters > 50;
```

![image](https://user-images.githubusercontent.com/16267906/72192120-195a9a80-33ca-11ea-88bd-80745474952c.png)

Could someone give me intuition on how to set bounds and epsilon?

Thanks 

While building the Postgres database, I'm getting anon_func.so No such file or directory. I tried looking in other folders too, but I couldn't find anything.

Need help!
@google-admin @willnorris @cdibona @dberlin @TedTed 

Here's the screenshot
![image](https://user-images.githubusercontent.com/43990819/71982572-c7d7d180-324a-11ea-8fff-d6475fde52a8.png)

In reviewing the source code, it is not clear which variables refer to private data and which refer to privitized data. I suggest that you clearly distinguish these in the source code, perhaps by using different variable naming conventions.
It appears that this project uses openssl/rand.h as its randomness source. Unfortunately, openssl's random number generator is frequently not sufficient for providing privacy protection many statistics. In our analysis, we have found that the only secure way to get private random numbers is to call the RDRAND instruction directly, or to use a pluggable random number generator that supports RDRAND. 

You can read more about the issue of OpenSSL and RDRAND [here](https://software.intel.com/en-us/blogs/2014/10/03/changes-to-rdrand-integration-in-openssl).

So it would be useful if you could support RDRAN directly.

ps: I just discovered this project. Congrats on putting it out!
https://github.com/google/differential-privacy/blob/master/differential_privacy/base/status_macros.h#L35

expr could be evaluated twice in the macro, which is problematic.
I get the following error when building for macOS Mojave

```
error: use of undeclared identifier 'SYNC_FILE_RANGE_WRITE'
        (void) sync_file_range(fd, 0, 0, SYNC_FILE_RANGE_WRITE);
                                         ^
1 warning and 1 error generated.
make[2]: *** [file_utils.o] Error 1
make[1]: *** [all-common-recurse] Error 2
make: *** [all-src-recurse] Error 2

_____ END BUILD LOGS _____
rules_foreign_cc: Build script location: bazel-out/darwin-fastbuild/bin/external/postgres/postgres/logs/Configure_script.sh
rules_foreign_cc: Build log location: bazel-out/darwin-fastbuild/bin/external/postgres/postgres/logs/Configure.log

INFO: Elapsed time: 58.778s, Critical Path: 57.91s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

Any ideas?
I'm attempting to add this as a package to conda-forge but am getting an error that I'm unsure of how to fix

I download the source from `https://github.com/google/differential-privacy/archive/master.tar.gz`

And a script that executes `bazel build differential_privacy/...`
Which returns the error 
```ERROR: circular symlinks detected
[start of symlink cycle]
/home/conda/staged-recipes/build_artifacts/google-differential-privacy_1571944966203/_build_env/share/terminfo/N/NCR260VT300WPP
[end of symlink cycle]
Internal error thrown during build. Printing stack trace: java.lang.IllegalStateException: //differential_privacy/algorithms:bounded-algorithm_test BuildConfigurationValue.Key[bb189304b1c2d885f4d7b75c97be3b24] false -> ErrorInfo{exception=com.google.devtools.build.lib.skyframe.FileSymlinkCycleException: Symlink cycle, rootCauses={FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp]}, cycles=[], isCatastrophic=false, rootCauseOfException=FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp], isDirectlyTransient=false, isTransitivelyTransient=false}
	at com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:548)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:365)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:368)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:212)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:121)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:253)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:83)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:482)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:750)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$1600(GrpcServerImpl.java:103)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:819)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

INFO: Elapsed time: 33.934s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (48 packages loaded, 1191 targets configured)
Internal error thrown during build. Printing stack trace: java.lang.IllegalStateException: //differential_privacy/algorithms:bounded-algorithm_test BuildConfigurationValue.Key[bb189304b1c2d885f4d7b75c97be3b24] false -> ErrorInfo{exception=com.google.devtools.build.lib.skyframe.FileSymlinkCycleException: Symlink cycle, rootCauses={FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp]}, cycles=[], isCatastrophic=false, rootCauseOfException=FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp], isDirectlyTransient=false, isTransitivelyTransient=false}
	at com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:548)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:365)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:368)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:212)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:121)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:253)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:83)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:482)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:750)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$1600(GrpcServerImpl.java:103)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:819)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
java.lang.IllegalStateException: //differential_privacy/algorithms:bounded-algorithm_test BuildConfigurationValue.Key[bb189304b1c2d885f4d7b75c97be3b24] false -> ErrorInfo{exception=com.google.devtools.build.lib.skyframe.FileSymlinkCycleException: Symlink cycle, rootCauses={FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp]}, cycles=[], isCatastrophic=false, rootCauseOfException=FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp], isDirectlyTransient=false, isTransitivelyTransient=false}
	at com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:548)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:365)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:368)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:212)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:121)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:253)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:83)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:482)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:750)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$1600(GrpcServerImpl.java:103)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:819)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
java.lang.IllegalStateException: //differential_privacy/algorithms:bounded-algorithm_test BuildConfigurationValue.Key[bb189304b1c2d885f4d7b75c97be3b24] false -> ErrorInfo{exception=com.google.devtools.build.lib.skyframe.FileSymlinkCycleException: Symlink cycle, rootCauses={FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp]}, cycles=[], isCatastrophic=false, rootCauseOfException=FILE:[/home/conda/.cache/bazel/_bazel_conda/09cbb6ed10102e8e294ff82d8a8f8c67/external/embedded_jdk]/[lib/terminfo/N/ncr260vt300wpp], isDirectlyTransient=false, isTransitivelyTransient=false}
	at com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:548)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:365)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:368)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:212)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:121)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:253)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:83)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:482)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:750)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$1600(GrpcServerImpl.java:103)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:819)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
FAILED: Build did NOT complete successfully (48 packages loaded, 1191 targets configured)
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 134, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File "/home/conda/.ci_support/build_all.py", line 124, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2339, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1487, in build
    cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 399, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 379, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/home/conda/staged-recipes/build_artifacts/google-differential-privacy_1571944966203/work/conda_build.sh']' returned non-zero exit status 37.
```

I'm not sure which is causing the error: conda-forge, bazel, or differential-privacy
Please advise if you've encountered this error before when building
Hello there, 
Would it be possible for you to generate a release for differential-privacy? Thanks!
